python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -644,7 +644,8 @@ def replicate(self, destSE, sourceSE=''):
         logger.info("Replicating file %s to %s" % (self.lfn, destSE))
         stdout = execute('replicateFile("%s", "%s", "%s")' % (self.lfn, destSE, sourceSE))
         if isinstance(stdout, dict) and stdout.get('OK', False) and self.lfn in stdout.get('Value', {'Successful': {}})['Successful']:
-            self.locations.append(destSE)
+            if destSE not in self.locations:
```

Comment:
As far as I am aware, if you try to replicate a file in DIRAC to an SE where it is already present, it will work just fine. It will return:
```python
{'Failed': {},
 'Successful': {'/gridpp/user/r/robot.general.gangaproject/tmp/test1.txt': {'register': 0,
                                                                            'replicate': 0}}}
```